### PR TITLE
[#434] Empty Collection State

### LINKED ISSUE #434: Empty collection state for tvOS
Show text indicating when a collection is empty (no products). Also allow the Focus Engine to focus onto that collection even if its empty.

### PR DESCRIPTION
**Pull request by EishanVijay**
Added label to indicate when a collection was empty (no products). Also allowed the Focus Engine to focus on an empty collection by providing it an empty product cell to focus to.

Resolves #434

@davidmuzi @bgulanowski

### CODE REVIEW

**Comment by EishanVijay** [comment]
Approved in PR #435

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift [hunk 1]
```diff
@@ -58,20 +62,26 @@ extension ProductsViewModel: UICollectionViewDataSource {
...
-        return self.products.count
+        return hasProducts ? self.products.count : 1
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ProductCollectionViewCell.reuseIdentifier, for: indexPath) as! ProductCollectionViewCell
-        let product = self.products[indexPath.row] as BUYProduct
-        if product.imagesArray().count > 0 {
-            cell.productImage.load(product.imagesArray().first, animateChange: true, completion: nil)
+        if hasProducts {
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift]: you can use an if let here.  `if let product = self.products[indexPath.row] as BUYProduct`

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift [hunk 2]
```diff
@@ -58,20 +62,26 @@ extension ProductsViewModel: UICollectionViewDataSource {
...
+            let product = self.products[indexPath.row] as BUYProduct
+            if product.imagesArray().count > 0 {
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift]: if let image = product.imagesArray.first

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift [hunk 3]
```diff
@@ -58,20 +62,26 @@ extension ProductsViewModel: UICollectionViewDataSource {
...
+                cell.productImage.load(product.imagesArray().first, animateChange: true, completion: nil)
+            } else {
+                cell.productImage.image = UIImage(named: "Placeholder")
+            }
+            let productItem = ProductItem(images: product.imagesArray())
+            let priceString = String(format: "%@", self.dataProvider.getCurrencyFormatter().string(from: product.minimumPrice)!)
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift]: no need to construct another string, remove: `String(format: "%@", `

**Pull request merged**